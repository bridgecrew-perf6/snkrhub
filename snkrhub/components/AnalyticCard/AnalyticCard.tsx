import {
  Text,
  Stack,
  Icon,
  VStack,
  Button,
  PresenceTransition
} from "native-base";
import { FontAwesome5, MaterialIcons } from '@expo/vector-icons'; 

// Types
type Props = {
    mainStat: number;
    prefix?: string;
    width?: string;
    subtext: string;
    cardIcon: React.ComponentProps<typeof FontAwesome5 | typeof MaterialIcons>
}

const AnalyticCard = (props: Props) => {
    return (
        <>
            <Stack 
                direction="row"
                alignItems={'center'}
                _light={{ bg: "gray.200" }}
                _dark={{ bg: "gray.800" }}
                mt="2"
                p="5"
                w="48%"
                borderRadius={6}
            >   
                <VStack>
                    {/* Icon */}
                    <Button
                        borderRadius="50"
                        px="0"
                        w={props.width}
                        bg="primary.500"
                    >
                        {props.cardIcon}
                    </Button>

                    {/* Main Stat */}
                    <PresenceTransition 
                        visible={props.mainStat || props.mainStat == 0 ? true : false} 
                        initial={{
                            opacity: 0
                        }} 
                        animate={{
                            opacity: 1,
                            transition: {
                                duration: 250
                            }
                        }}
                    >
                        <Text
                            pt="3"
                            fontSize="lg"
                            fontWeight="bold"
                            _light={{
                                color: "gray.700",
                            }}
                            _dark={{
                                color: "gray.300", 
                            }}
                        >
                            {props.prefix}{props.mainStat}   
                        </Text>
                    </PresenceTransition>
                    

                    {/* Subtext */}
                    <Text
                        fontSize="xs"
                        _light={{
                            color: "gray.700",
                        }}
                        _dark={{
                            color: "gray.300", 
                        }}
                    >
                        {props.subtext ? 
                            <>
                                {props.subtext}
                            </>
                            : 
                            null
                        }  
                    </Text>
                </VStack>
            </Stack>
        </>
    )
}

export default AnalyticCard