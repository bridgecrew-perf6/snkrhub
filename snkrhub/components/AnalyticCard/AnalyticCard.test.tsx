import * as React from 'react';
import renderer from 'react-test-renderer';

import { FontAwesome5 } from '@expo/vector-icons';
import { Icon } from "native-base";
import AnalyticCard from './AnalyticCard';
import { NativeBaseProvider } from 'native-base';

// test that inventory item accepts correct props
it(`analytics card accepts the correct props`, () => {
  const props = {
    mainStat: 200, 
    width: '38%',
    subtext: 'Inventory sold',
    cardIcon: <Icon as={FontAwesome5} name="shopping-bag" size="3" color="white" ml="0.2"/>
  }

  const tree = renderer.create(
    <NativeBaseProvider>
      <AnalyticCard {...props} />
    </NativeBaseProvider>
  ).toJSON()

  expect(tree).toMatchSnapshot();
});
