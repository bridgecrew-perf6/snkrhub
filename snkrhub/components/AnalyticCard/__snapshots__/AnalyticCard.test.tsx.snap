// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analytics card accepts the correct props 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
